From 5f86c406bb76d096241c82519bd3bff81032fdac Mon Sep 17 00:00:00 2001
From: affggh <879632264@qq.com>
Date: Mon, 2 Sep 2024 19:46:57 +0800
Subject: [PATCH] tools: Fix missing struct member policydb_write


diff --git a/kernel/include/preset.h b/kernel/include/preset.h
index b63772d..ea4bd61 100644
--- a/kernel/include/preset.h
+++ b/kernel/include/preset.h
@@ -119,6 +119,7 @@ struct patch_symbol
             uint64_t avc_denied;
             uint64_t slow_avc_audit;
             uint64_t input_handle_event;
+            uint64_t policydb_write;
         };
         char _cap[PATCH_SYMBOL_LEN];
     };
-- 
2.45.1

